cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolationDetector.java

```diff
@@ -228,9 +228,9 @@ public void run() {
       _provisionResponse = provisionResponse;
       if (_isProvisionerEnabled) {
         // Right-size the cluster (if needed)
-        boolean isRightsized = _provisioner.rightsize(_provisionResponse.recommendationByRecommender());
-        if (isRightsized) {
-          LOG.info("Actions have been taken on the cluster towards rightsizing.");
+        ProvisionerState isRightsized = _provisioner.rightsize(_provisionResponse.recommendationByRecommender(), new RightsizeOptions());
+        if (isRightsized.summary() != null) {
+          LOG.info(isRightsized.summary());
```

Comment:
1. Adding the override toString() method in ProvisionerState
2. Updating to return null when no actions are taken (in the Provisioner and JavaDocs)
3. Modifying the provisioners to return null so that we can avoid logging when no action is taken.